src/main/src/main/java/org/geoserver/catalog/util/LegacyCatalogImporter.java

```diff
@@ -282,7 +282,8 @@ void importCatalog(File catalogFile) throws FileNotFoundException,
         reader.read(Files.asResource(catalogFile));
 
         // build all the catalog objects that can be read from the catalog.xml file
-        importNamespaces(factory, reader.namespaces());
+        importNamespaces(factory, reader.namespaces(), false);
+        importNamespaces(factory, reader.isolatedNamespaces(), true);
```

Comment:
A true legacy catalog won't have isolated workspaces. I'm guessing this support is there due to tests (they are still writing catalogs using the old structure, aren't they?)

Reply:
I understand